src/xdist/remote.py

```diff
@@ -241,15 +241,17 @@ def pytest_collection_modifyitems(
         # add the group name to nodeid as suffix if --dist=loadgroup
         if config.getvalue("loadgroup"):
             for item in items:
-                mark = item.get_closest_marker("xdist_group")
-                if not mark:
+                gnames: set[str] = set()
+                for mark in item.iter_markers("xdist_group"):
+                    name = (
```

Comment:
The logic for getting the name of the mark seems involved enough to make it a function 

A fun hack would be
```
def group_name_of(name,*k,**kw):
  return name

groups = {group_name_of(*mark.args, **mark.kwargs) for mark in item.iter....}
```